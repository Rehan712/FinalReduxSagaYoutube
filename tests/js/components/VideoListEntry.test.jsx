import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import VideoListEntry from '../../../src/js/components/VideoListEntry';

Enzyme.configure({ adapter: new Adapter() });

describe('Test Video List Entry', () => {
	const video = {
		id: { videoId: 1, playlistId: 2 },
		snippet: {
			title: 'naruto',
			description: 'ninja',
			thumbnails: {
				default: {
					url: 'naruto.com'
				}
			}
		}
	};
	const handleSelectedVideo = jest.fn();
	const wrapper = shallow(
		<VideoListEntry
			key={video.id.videoId}
			video={video}
			handleSelectedVideo={handleSelectedVideo}
		/>
	);
	it('checks the rendering of the component', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('checks the onclick callback', () => {
		wrapper.find('.media-heading').simulate('click');
	});
});
