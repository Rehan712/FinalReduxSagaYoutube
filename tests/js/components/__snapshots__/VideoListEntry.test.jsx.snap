// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Video List Entry checks the rendering of the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    video={
      Object {
        "id": Object {
          "playlistId": 2,
          "videoId": 1,
        },
        "snippet": Object {
          "description": "ninja",
          "thumbnails": Object {
            "default": Object {
              "url": "naruto.com",
            },
          },
          "title": "naruto",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="media"
      >
        <div
          className="media-left media-middle"
        >
          <img
            alt=""
            className="media-object"
            src="naruto.com"
          />
        </div>
        <div
          className="media-body"
        >
          <button
            className="media-heading"
            onClick={[Function]}
          >
            naruto
          </button>
          <p
            className="media-description"
          >
            ninja
          </p>
        </div>
      </div>,
      "className": "videoListEntry",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="media-left media-middle"
          >
            <img
              alt=""
              className="media-object"
              src="naruto.com"
            />
          </div>,
          <div
            className="media-body"
          >
            <button
              className="media-heading"
              onClick={[Function]}
            >
              naruto
            </button>
            <p
              className="media-description"
            >
              ninja
            </p>
          </div>,
        ],
        "className": "media",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              className="media-object"
              src="naruto.com"
            />,
            "className": "media-left media-middle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "media-object",
              "src": "naruto.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="media-heading"
                onClick={[Function]}
              >
                naruto
              </button>,
              <p
                className="media-description"
              >
                ninja
              </p>,
            ],
            "className": "media-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "naruto",
                "className": "media-heading",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "naruto",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ninja",
                "className": "media-description",
              },
              "ref": null,
              "rendered": "ninja",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="media"
        >
          <div
            className="media-left media-middle"
          >
            <img
              alt=""
              className="media-object"
              src="naruto.com"
            />
          </div>
          <div
            className="media-body"
          >
            <button
              className="media-heading"
              onClick={[Function]}
            >
              naruto
            </button>
            <p
              className="media-description"
            >
              ninja
            </p>
          </div>
        </div>,
        "className": "videoListEntry",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="media-left media-middle"
            >
              <img
                alt=""
                className="media-object"
                src="naruto.com"
              />
            </div>,
            <div
              className="media-body"
            >
              <button
                className="media-heading"
                onClick={[Function]}
              >
                naruto
              </button>
              <p
                className="media-description"
              >
                ninja
              </p>
            </div>,
          ],
          "className": "media",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                className="media-object"
                src="naruto.com"
              />,
              "className": "media-left media-middle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "media-object",
                "src": "naruto.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="media-heading"
                  onClick={[Function]}
                >
                  naruto
                </button>,
                <p
                  className="media-description"
                >
                  ninja
                </p>,
              ],
              "className": "media-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "naruto",
                  "className": "media-heading",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "naruto",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ninja",
                  "className": "media-description",
                },
                "ref": null,
                "rendered": "ninja",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
